msf-amp-example/README.md

```diff
@@ -0,0 +1,68 @@
+## Sample application: Amazon Managed Service for Apache Flink and Amazon Managed Prometheus
```

Comment:
We shouldn't be putting company specific examples here. Should we consider using a non-MSF example here instead, and move this MSF-specific one elsewhere?

Reply:
Replaced the example with a new one, using Flink 1.17.2 and with nothing MSF specific.
The new example optionally demonstrates how to configure the request signer to target AMP (the only supported request signer at the moment)

Reply:
Updated to 1.18.1